ledger/src/shred/wire.rs

```diff
@@ -323,6 +323,15 @@ pub fn resign_packet(packet: &mut PacketRefMut, keypair: &Keypair) -> Result<(),
 
             packet.set_buffer(buffer);
 
+            Ok(())
+        }
+        PacketRefMut::WithClientId(packet) => {
```

Comment:
I don't understand when batch has or has not the Pubkey. Is it only for votes?

Reply:
It is controlled by QuicStreamerConfig's send_client_id. The consumer use that flag to indicate its interest. This is for designed for alpenglow to receive the pubkey of the client. The alpenglow code will need to be modified to turn on that flag. In addition, it can be enhanced in the future to support feedback mechanism for regular TPU.

Reply:
This branch should never be taken in turbine. I'd rather we drop the packet if we hit this variant.

Reply:
No longer relevant as we are sending via Packet::Meta